debezium-core/src/main/java/io/debezium/relational/history/TableChanges.java

```diff
@@ -199,98 +139,26 @@ else if (!table.equals(other.table)) {
-                document.setNumber("nativeType", column.nativeType());
-            }
-
-            document.setString("typeName", column.typeName());
-            document.setString("typeExpression", column.typeExpression());
-            document.setString("charsetName", column.charsetName());
-
-            if (column.length() != Column.UNSET_INT_VALUE) {
-                document.setNumber("length", column.length());
-            }
-
-            column.scale().ifPresent(s -> document.setNumber("scale", s));
-
-            document.setNumber("position", column.position());
-            document.setBoolean("optional", column.isOptional());
-            document.setBoolean("autoIncremented", column.isAutoIncremented());
-            document.setBoolean("generated", column.isGenerated());
-
-            return document;
-        }
-
-        private static Table fromDocument(TableId id, Document document) {
-            TableEditor editor = Table.editor()
-                    .tableId(id)
-                    .setDefaultCharsetName(document.getString("defaultCharsetName"));
-
-            document.getArray("columns")
-                    .streamValues()
-                    .map(Value::asDocument)
-                    .map(v -> {
-                        ColumnEditor columnEditor = Column.editor()
-                                .name(v.getString("name"))
-                                .jdbcType(v.getInteger("jdbcType"));
-
-                        Integer nativeType = v.getInteger("nativeType");
-                        if (nativeType != null) {
-                            columnEditor.nativeType(nativeType);
-                        }
-
-                        columnEditor.type(v.getString("typeName"), v.getString("typeExpression"))
-                                .charsetName(v.getString("charsetName"));
-
-                        Integer length = v.getInteger("length");
-                        if (length != null) {
-                            columnEditor.length(length);
-                        }
-
-                        Integer scale = v.getInteger("scale");
-                        if (scale != null) {
-                            columnEditor.scale(scale);
-                        }
-
-                        columnEditor.position(v.getInteger("position"))
-                                .optional(v.getBoolean("optional"))
-                                .autoIncremented(v.getBoolean("autoIncremented"))
-                                .generated(v.getBoolean("generated"));
-
-                        return columnEditor.create();
-                    })
-                    .forEach(editor::addColumn);
+    /**
+     * The interface that defines conversion of {@code TableChanges} into a serialized format for
+     * persistent storage or delivering as a message.
+     *
+     * @author Jiri Pechanec
+     *
+     * @param <T> target type
+     */
+    public static interface TableChangesSerializer<T> {
```

Comment:
Can you clarify why there are two implementations?

Reply:
one is for internal representation of schema history - directly written into the JSON.
The other is public facing structure for external consuming - can be JSON/Avro according to the users needs and also public and private format will not affect each other.

Reply:
Ok, thanks for clarifying.